### DISCUSSION TITLE
Attestation endpoint calls keep failing with 422 code

### DISCUSSION THREAD

**Discussion by jakub-jarzabek** [WalletConnect/walletconnect-monorepo#2329] [Developer Support]
I've been using web3modal in an application for some time now and it worked fine, but recently API calls for `https://verify.walletconnect.com/attestation` started to return 422 status code, which made connection and integration with wagmi hooks unusable.

In dev tools network tab I see constantly 10+ failed calls to `/attestation` endpoint with status 422 and response `Failed to deserialize the JSON body into the target type: attestationId: invalid type: map, expected a string at line 1 column 17`.

Here are version of packages that I'm using:
```
 "@web3modal/ethereum": "2.3.7",
 "@web3modal/react": "2.3.7",
 "wagmi": "0.12.12",
  ```

**Comment by xzilja** [comment]
@jakub-jarzabek are you still seeing these issues on latest dependencies? Ensure you .lock files are also up to date

**Reply by jakub-jarzabek** [reply]
yes, I still see those issues with the latest dependencies, but setting version property to `1` when instantiating connectors somehow solved that issue.

```diff
const wagmiClient = createClient({
  autoConnect: true,
  connectors: w3mConnectors({
    projectId,
    chains,
-   version: 2,
+   version: 1,
  }),
  provider,
})
```

**Reply by xzilja** [reply]
Yeh, v1 doesn't use attestation

**Reply by arein** [reply]
@jakub-jarzabek can you share a screenshot and the cURL for the failing request please? 

In Chrome -> dev tools -> network -> right click on failed request -> copy -> copy as curl 

🙏

**Reply by jakub-jarzabek** [reply]
@arein Here are screenshots 
<img width="828" alt="Screenshot 2023-05-10 at 13 41 11" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/57012225/b1a9df8a-784a-45d6-8e0e-00404501bc9f">
<img width="835" alt="Screenshot 2023-05-10 at 13 41 32" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/57012225/aa246514-7eb9-4263-b446-9c9dae036abe">

Here is cURL request before connecting wallet:
```bash
curl 'https://verify.walletconnect.com/attestation' \
  -H 'authority: verify.walletconnect.com' \
  -H 'accept: */*' \
  -H 'accept-language: en-US,en;q=0.9' \
  -H 'content-type: application/json' \
  -H 'origin: https://verify.walletconnect.com' \
  -H 'referer: https://verify.walletconnect.com/1f0b8f82732fb2c7906925132069f6fc' \
  -H 'sec-ch-ua: "Google Chrome";v="113", "Chromium";v="113", "Not-A.Brand";v="24"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: same-origin' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36' \
  --data-raw '{"attestationId":{"target":"phantom-inpage","data":"SYN"},"origin":"https://verify.walletconnect.com"}' \
  --compressed
```
And here is one after connecting wallet:
```bash
curl 'https://verify.walletconnect.com/attestation' \
  -H 'authority: verify.walletconnect.com' \
  -H 'accept: */*' \
  -H 'accept-language: en-US,en;q=0.9' \
  -H 'content-type: application/json' \
  -H 'origin: https://verify.walletconnect.com' \
  -H 'referer: https://verify.walletconnect.com/1f0b8f82732fb2c7906925132069f6fc' \
  -H 'sec-ch-ua: "Google Chrome";v="113", "Chromium";v="113", "Not-A.Brand";v="24"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: same-origin' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36' \
  --data-raw '{"attestationId":{"target":"metamask-inpage","data":{"name":"metamask-provider","data":{"method":"metamask_chainChanged","params":{"chainId":"0x89","networkVersion":"137"}}}},"origin":"https://verify.walletconnect.com"}' \
  --compressed
```

**Reply by ganchoradkov** [reply]
Hey @jakub-jarzabek, thanks for the logs but I'm unable to reproduce it on our example dapp nor wagmi docs. The requests are being processed correctly.
https://react-app.walletconnect.com/
https://wagmi.sh/examples/connect-wallet 

Could you provide a bit more info about your project, bundlers, browser used so we can reproduce it?

**Reply by ganchoradkov** [reply]
The `attestationId` should be the hash of the payload 
https://github.com/WalletConnect/walletconnect-monorepo/blob/v2.0/packages/sign-client/src/controllers/engine.ts#L438

Also wondering why your request is in play text.

**Reply by jakub-jarzabek** [reply]
Hey @ganchoradkov I'm using wallet connect in Next.js (version 12.3.0) app and Chrome (version Version 113.0.5672.92), am using `w3mConnectors` and `SafeConnector` as my available connectors:
```ts
 const { provider } = configureChains(chains, [
        w3mProvider({
            projectId: process.env.NEXT_PUBLIC_WALLET_CONNECT_PROJECT_ID,
        }),
    ]);

    const safeConnector = new SafeConnector({
        chains: [mainnet, polygon],
        options: {
            allowedDomains: [/gnosis-safe.io$/, /app.safe.global$/],
            debug: true,
        },
    });

    const walletConnectConnectors = w3mConnectors({
        projectId: process.env.NEXT_PUBLIC_WALLET_CONNECT_PROJECT_ID,
        chains,
        
        version: 1,
    });

    const wagmiClient = createClient({
        connectors: [...walletConnectConnectors, safeConnector],
        provider,
    });
    const ethereumClient = new EthereumClient(wagmiClient, chains);
```
Also, I was trying to create a new project and set it up with a new project Id, but it still didn't work.

**Comment by chybisov** [comment]
<img width="960" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/18644653/79d81551-6721-46c4-a982-ca61477179cf">

@ganchoradkov, we have a similar issue while trying to upgrade to WalletConnect v2.

Please check this PR https://github.com/lifinance/widget/pull/89

To run the example: 
1. Run `yarn` in the root 
2. Open `packages/widget-playground` in a terminal and run `yarn dev`

Thank you!

**Comment by tiger-525** [comment]
Hello.
I have a same issue when I am trying to connect walletconnect connector by using wagmi as follow.
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/55833057/5c080dec-83b6-4fc4-9342-5d720fbd7273)

I think the problem may be caused by the "project id" setting.
May I simply create a project on "cloud.walletconnect.com" to get a "project id"?
Don't need any other additional settings like "Explorer" or "Settings"?

**Comment by ganchoradkov** [comment]
Hey @jakub-jarzabek @chybisov @bin0316 we've identified the issue and are rolling out a fix today.
It's a backend fix so no SDK update is required. The issue is caused by interaction between the metamask inpage-provider and the Verify Enclave

**Reply by chybisov** [reply]
Cool, thank you 🙂
